torchdata/datapipes/iter/util/prefetcher.py

```diff
@@ -131,10 +132,129 @@ def reset(self):
 
     def pause(self):
         if self.thread is not None:
+            assert self.prefetch_data is not None
             self.prefetch_data.run_prefetcher = False
 
     def resume(self):
         if self.thread is not None and (
             not self.prefetch_data.stop_iteration or len(self.prefetch_data.prefetch_buffer) > 0
         ):
+            assert self.prefetch_data is not None
             self.prefetch_data.run_prefetcher = True
+
+
+def pin_memory_fn(data, device=None):
```

Comment:
Should we keep this public and add it to `__init__.py` somewhere else for broader public usage and extension? Similar to how `collate_fn` is in Core.

Reply:
Make sense.